### DISCUSSION TITLE
Newlines & VSCode

### DISCUSSION THREAD

**Discussion by ciroiriarte** [netbox-community/devicetype-library#1361] [General]
Hello!, can anybody provide some tips about how to work with this repo & VSCode on Windows?.

For *reasons* I have to work with that combination and I'm struggling with new line formatting and yamllint. The weird think is that either way I end up with the wrong number of trailing lines on the new files, either too many or too few.

Tried setting new lines to "LF", same output. I have a PR stuck because of this.

**Comment by seaburger** [comment]
Ran into the same issue with this.

The issue is that the YAML formatter being used ([pre-commit-hook-yamlfmt](https://github.com/jumanjihouse/pre-commit-hook-yamlfmt)) rewrites the files.  Python uses the OS default line ending format by default when it writes files.  So when this runs on Windows, it will change the YAML file from `\n` (LF) line endings to `\r\n` (CRLF).

The YAML linter to validate formatting ([yamllint](https://github.com/adrienverge/yamllint)) is configured to require all YAML files in this project to use `\n` newlines.

I've submitted a PR for the YAML formatter being used to allow overriding this behavior (https://github.com/jumanjihouse/pre-commit-hook-yamlfmt/pull/57).  If that PR gets accepted, then this project can update the `yamlfmt` parameters its using to fix this. 


---
There is a separate 2nd issue on Windows that should be fixed by another PR.  This one prevents the `yamlfmt` command from running at all.
Issue: (https://github.com/jumanjihouse/pre-commit-hook-yamlfmt/issues/25) (https://github.com/jumanjihouse/pre-commit-hook-yamlfmt/issues/32)
PR: (https://github.com/jumanjihouse/pre-commit-hook-yamlfmt/pull/42)

You can manually just go into the `yamlfmt` file in `.cache` folder and change the single `\` to a double `\\` in the `python.exe` path right at the top of the file.
